src/Features/CSharp/Portable/PopulateSwitch/CSharpPopulateSwitchCodeFixProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.PopulateSwitch
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.PopulateSwitch), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.ImplementInterface)]
+    internal class CSharpPopulateSwitchCodeFixProvider : AbstractPopulateSwitchCodeFixProvider<SwitchStatementSyntax>
+    {
+        protected override SyntaxNode GetSwitchStatementNode(SyntaxNode root, TextSpan span)
```

Comment:
Sorry @Hosch250 , i don't think i was clear.  ```AbstractPopulateSwitchCodeFixProvider``` shoudl define GetSwitchStatementNode like so:  ```protected abstract TSwitchBlockSyntax GetSwitchStatementNode```.  Then, when you override it yourself it will be strongly typed to return a SwitchStatementSyntax